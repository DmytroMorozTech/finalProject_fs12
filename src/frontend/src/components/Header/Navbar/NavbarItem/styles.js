import { makeStyles } from '@material-ui/core/styles'

export default makeStyles((theme) => ({
  menuItem: {
    display: 'flex',
    flexDirection: 'column',
    cursor: 'pointer',
    paddingTop: theme.spacing(1),
    transition: 'all 0.4s ease',
    lineHeight: theme.typography.h4.lineHeight,
    '&:hover': {
      color: theme.palette.common.black
    }
  },
  icons: {
    width: theme.spacing(7),
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    fontSize: theme.typography.icons.header.fontSize
  },
  title: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    width: theme.spacing(16),
    height: theme.spacing(3),
    '& > span': {
      fontSize: theme.typography.h6.fontSize,
      fontWeight: theme.typography.header.fontWeight,
      marginBottom: theme.spacing(1)
    }
  },
  itemPrimary: {
    textDecoration: 'none',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    color: theme.palette.grey[700],
    '&:hover': {
      color: theme.palette.common.black,
      fontWeight: 'bold'
    }
  },
  itemPrimaryActive: {
    transition: 'all 0.4s ease',
    '& > div': {
      color: theme.palette.common.black
    },
    '& hr': {
      width: theme.spacing(16),
      transition: 'width 0.4s ease-in'
    }
  },
  arrow: {
    position: 'relative',
    top: -theme.spacing(0.5),
    left: -theme.spacing(0.5)
  },
  line: {
    width: '0px',
    height: '1.5px',
    border: '0',
    marginTop: '0',
    marginBottom: '0',
    backgroundColor: theme.palette.common.black,
    transition: 'width 0.4s ease-out'
  },
  badge: {
    '& > span': {
      top: 7,
      right: 10
    }
  }
}))
