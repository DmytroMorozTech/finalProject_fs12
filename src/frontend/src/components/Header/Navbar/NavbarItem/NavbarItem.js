import React from 'react'
import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown'
import { Hidden } from '@material-ui/core'
import styles from './styles'
import { NavLink } from 'react-router-dom'
import SimpleMenu from '../../../../shared/PopupMenu/PopupMenu'
import UserData from '../../../UserData/UserData'
import * as actions from '../../../../redux/Message/messageActionTypes'
import {useDispatch} from 'react-redux'

const NavbarItem = ({ Icon, title, arrow, toggleMenu, to, exact = true }) => {
  const classes = styles()
  const dispatch = useDispatch()

  const notLink = (e) => {
    e.preventDefault()
  }

  const renderMenuItem = (
    <NavLink className={classes.itemPrimary}
      onClick={(e) => {
        toggleMenu && notLink(e)
        dispatch({ type: actions.SELECTED_CHAT, payload: false })
      }}
      activeClassName={classes.itemPrimaryActive}
      exact={exact}
      to={to}
    >
      <div className={classes.icons}>{Icon}</div>
      <Hidden xsDown>
        <div className={classes.title}>
          <span>{title}</span>
          {arrow && <ArrowDropDownIcon fontSize="inherit" className={classes.arrow} />}
        </div>
        <hr className={classes.line} />
      </Hidden>
    </NavLink>
  )

  const renderMenu = toggleMenu
    ? <SimpleMenu menuItem={renderMenuItem} userData={<UserData />} />
    : renderMenuItem

  return (
    <div className={classes.menuItem}>
      {renderMenu}
    </div>
  )
}

export default NavbarItem
