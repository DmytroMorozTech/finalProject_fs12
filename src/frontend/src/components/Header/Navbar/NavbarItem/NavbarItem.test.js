import { shallow } from 'enzyme'
import * as redux from 'react-redux'
import React from 'react'
import NavbarItem from './NavbarItem'
import HomeRoundedIcon from '@material-ui/icons/HomeRounded'

jest.mock('./styles', () => () => ({}))

const setUp = (props) => shallow(<NavbarItem {...props}/>)

describe('should render NavbarItem component', () => {
  let spyOnUseSelector
  let spyOnUseDispatch
  let mockDispatch

  let Icon
  let title
  let arrow
  let toggleMenu
  let badge
  let to

  let component

  beforeEach(() => {
    spyOnUseSelector = jest.spyOn(redux, 'useSelector')
    spyOnUseSelector.mockReturnValue(false)
    spyOnUseDispatch = jest.spyOn(redux, 'useDispatch')
    mockDispatch = jest.fn()
    spyOnUseDispatch.mockReturnValue(mockDispatch)

    Icon = <HomeRoundedIcon fontSize='inherit' onClick={() => mockDispatch}/>
    title = 'Home'
    arrow = false
    toggleMenu = false
    badge = false
    to = '/home'

    component = setUp({Icon, title, arrow, toggleMenu, badge, to})
  })

  afterEach(() => {
    jest.restoreAllMocks()
  })

  it('to match snapshot', () => {
    expect(component).toMatchSnapshot()
  })

  it('should work SearchBarIconButton', () => {
    expect(mockDispatch).toHaveBeenCalledTimes(0)
    component.find('NavLink').simulate('click')
    expect(mockDispatch).toHaveBeenCalledTimes(1)
  })
})
