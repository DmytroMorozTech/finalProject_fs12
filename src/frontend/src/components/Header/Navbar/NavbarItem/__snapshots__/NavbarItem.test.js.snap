// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NavbarItem component to match snapshot 1`] = `
<div>
  <NavLink
    exact={true}
    onClick={[Function]}
    to="/home"
  >
    <div>
      <Memo(ForwardRef(HomeRoundedIcon))
        fontSize="inherit"
        onClick={[Function]}
      />
    </div>
    <Hidden
      xsDown={true}
    >
      <div>
        <span>
          Home
        </span>
      </div>
      <hr />
    </Hidden>
  </NavLink>
</div>
`;
